mantidproject/mantid
# SANSILL: Expose CalculateEfficiency thresholds to user

**Description of work.**

The issue related to sensitivity maps containing NaNs (+- inf) stemmed from too restrictive defaults for maximum threshold of possible efficiency in the `CalculateEfficiency` algorithm.  The proposed solution exposes the `MinThreshold` and `MaxThreshold` parameters of the `CalculateEfficiency` to the user of `SANSILL` family of algorithms. Two relevant tests were added to test that the values are now finite.

**To test:**

1. Run the following line, ensure you have UnitTest data in your search path:

`SANSILLReduction(Runs="ILL/D16/022846", ProcessAs="Water", OutputSensitivityWorkspace="sens", OutputWorkspace="_", MaxThreshold=5)`

2. The `sens` workspace should only contain numbers in the range 0 to 5 and there should be no NaNs.



Fixes #35028. 

Release mention added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
